### PR TITLE
Dropdown Component - Initialize when mounted

### PR DESCRIPTION
**Pull request by alextrastero**
Related to https://github.com/react-materialize/react-materialize/issues/74

Added a idx ref and initialised dropdown when mounted

This fixes only the dropdown, more to come.

### CODE REVIEW

**Comment by alextrastero** [comment]
```
<Dropdown options={ { hover: true } } trigger={ this.renderTrigger() }>
```

**Comment by alextrastero** [comment]
WIP, need to check if it still works

**Comment by wedneyyuri** [comment]
@alextrastero Have you thought about memory leak possibilities since we are not removing the event listeners?

https://facebook.github.io/react/docs/component-specs.html#unmounting-componentwillunmount

**Comment by alextrastero** [comment]
Good catch @wedneyyuri, i've update the PR, ready for codereview

**Comment by wedneyyuri** [comment]
@alextrastero What do you think about removing the `beloworigin` option from props/defaultProps? I think we can pass this value inside the `options` props. 

Everything seems good to me, a good work :)

**Comment by alextrastero** [comment]
@wedneyyuri yeah I thought about it, but IMO it's a must to have it `beloworigin` as default. but we still want to keep it customizable

File: src/Dropdown.js [hunk 1]
```diff
@@ -1,50 +1,66 @@
...
 
-class Dropdown extends React.Component {
+class Dropdown extends Component {
   constructor(props) {
     super(props);
     this.renderTrigger = this.renderTrigger.bind(this);
   }
 
+  componentDidMount() {
+    const options = this.props.options || {};
```
Comment (wedneyyuri) [inline src/Dropdown.js]: What do you think about it?

```
const options = $.extend({
  belowOrigin: true
}, this.props.options);
```

File: src/Dropdown.js [hunk 2]
```diff
@@ -1,50 +1,66 @@
...
+   * Options hash for the dropdown
    */
-  overorigin: React.PropTypes.bool,
+  options: PropTypes.object
 };
 
 Dropdown.defaultProps = {
-  overorigin: false,
-}
+  beloworigin: true
```
Comment (wedneyyuri) [inline src/Dropdown.js]: We can add our default value for options here:
`options: { }`

**Comment by wedneyyuri** [comment]
My point is make all the customisation inside the same `options` object to keep the original behaviour from materializeCSS.

Another possibility is using destructors (maybe a breaking change):

`<Dropdown inDuration={300} hover={true} />`

```
componentDidMount() {
  const options = this.props;
  ...
}
```

**Comment by alextrastero** [comment]
Yeah, the problem with this is that trigger, className or other props would also be passed to the dropdown

**Comment by alextrastero** [comment]
I mean, if what we want is to keep the functionality like materialize then we should remove the beloworigin is just pass the `options={ hover: true, beloworigin: true }` as a prop

**Comment by wedneyyuri** [comment]
I agree

**Comment by alextrastero** [comment]
@wedneyyuri check out the latest commit :)

**Comment by wedneyyuri** [comment]
@alextrastero Seems good to me.

**Comment by alextrastero** [comment]
The documentation should be update to reflect these changes
 - beloworigin is not a proptype anymore
 - new proptype to be used like so:
```JS
      <Dropdown
        trigger={<span>Trigger me</span>}
        options={
          {
            inDuration: 123,
            outDuration: 321,
            constrain_width: true,
            hover: true,
            gutter: 10,
            belowOrigin: true,
            alignment: 'right'
          }
        }
      >
```

**Comment by isaiah** [comment]
:rocket:

**Pull request merged**